cobigen-cli/src/main/java/com/devonfw/cobigen/cli/commands/UpdateCommand.java

```diff
@@ -0,0 +1,171 @@
+
+                        if (Integer.parseInt(localVersion[ver]) < Integer.parseInt(centralversionValue[ver])) {
+                            count++;
+                            String requireupdate = lclDependency.getArtifactId();
+                            logger.info("(" + (count) + ")" + requireupdate + " , " + lclDependency.getVersion());
+
+                        }
+                    }
+
+                }
+
+            }
+            logger.info(
+                "Here are the components that can be updated, which ones do you want to  update? Please list the");
+            ArrayList<String> userInputPluginForUpdate = new ArrayList<>();
+            for (String userArtifact : GenerateCommand.getUserInput().split(",")) {
+                userInputPluginForUpdate.add(userArtifact);
+            }
+            for (int j = 0; j < userInputPluginForUpdate.size(); j++) {
+                String currentSelectedArtifact = userInputPluginForUpdate.get(j);
+
+                String digitMatch = "\\d+";
+                // If given generable artifact is Integer
+                if (currentSelectedArtifact.matches(digitMatch)) {
+                    int selectedArtifactNumber = Integer.parseInt(currentSelectedArtifact);
+                    int index = selectedArtifactNumber - 1;
+                    // We need to generate all
+                    if (selectedArtifactNumber == 0) {
+                        logger.info("(0) All");
+                    }
+                    logger.info("Updating the following components:");
+
+                    File cpFile = rootCLIPath.resolve(MavenConstants.CLASSPATH_OUTPUT_FILE).toFile();
+                    cpFile.deleteOnExit();
+                    if (pomFile.exists()) {
+                        model = reader.read(new FileReader(pomFile));
+                    }
+                    // updating artificial pom
+                    for (Dependency lclDependency : localPomDepedency) {
+                        String[] localVersion = lclDependency.getVersion().split("\\.");
+                        if (!lclDependency.getArtifactId().equals("freemarker")
+                            && !lclDependency.getArtifactId().equals("mmm-code-java-parser")
+                            && !lclDependency.getArtifactId().equals("mmm-code-base")
+                            && !lclDependency.getArtifactId().equals("mmm-code-java-maven")
+                            && !lclDependency.getArtifactId().equals("mmm-util-core")) {
+                            String centralMavenversion =
+                                PluginUpdateUtil.checkLatestMavenVersion(lclDependency.getArtifactId());
+
+                            String[] centralversionValue = centralMavenversion.split("\\.");
+                            for (int ver = 0; ver < localVersion.length; ver++) {
+
+                                if (Integer.parseInt(localVersion[ver]) < Integer.parseInt(centralversionValue[ver])) {
```

Comment:
In these lines you have incorrectly duplicated the logic to check which component should be updated. The correct logic flow is the following:

1. Read pom to check which dependencies can be updated. (correct)
2. Print to the user the dependencies that can be updated, (correct)
3. User selects which dependencies to update e.g.`1,4,6`. (correct)
4. **Update  only** the dependencies that the user wants to update. (incorrect in your code, you are updating all of them)

Please change your logic to the correct one.